### PR TITLE
DropDownButton Demo: Set Button Template for DropDownButton

### PR DESCRIPTION
**Pull request by jdvictoria**

### CODE REVIEW

File: apps/demos/Demos/DropDownButton/Overview/jQuery/styles.css [hunk 1]
```diff
@@ -20,3 +20,42 @@
...
+.dx-button-title {
+  width: 87px;
+  height: 18px;
+  font-weight: 400;
+  font-size: 14px;
+  line-height: 18px;
+}
+
+.dx-button-row {
+  width: 62px;
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/DropDownButton/Overview/jQuery/styles.css]: according to figma we should add opacity: 0.6 here

File: apps/demos/Demos/DropDownButton/Overview/Vue/App.vue [hunk 2]
```diff
@@ -8,31 +8,44 @@
...
-            text="Sandra Johnson"
-            icon="../../../../images/gym/coach-woman.png"
-            display-expr="name"
-            key-expr="id"
-            @button-click="onButtonClick"
-            @item-click="onItemClick"
-          />
+              :split-button="true"
+              :use-select-mode="false"
+              :items="data.profileSettings"
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/DropDownButton/Overview/Vue/App.vue]: Please don't add unnecessary changes. it is harder to review the PR. If you'd like to make changes to the indentations, it's better to discuss it and do it in another PR

File: apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.css [hunk 3]
```diff
@@ -16,3 +16,57 @@
   cursor: pointer;
   font-size: 18px;
 }
+
+::ng-deep #text {
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.css]: What is the purpose of this? Is it necessary?

**Comment by jdvictoria** [comment]
[Demos: DropDownButton template](https://trello.com/c/LMFG28Af/8461-demos-dropdownbutton-template)

File: apps/demos/Demos/DropDownButton/Overview/Vue/App.vue [hunk 4]
```diff
@@ -171,22 +186,74 @@ function onColorClick(clickedColor) {
...
+  width: 24px;
+  height: 24px;
+}
+
+.color {
+  cursor: pointer;
+  font-size: 18px;
+}
 
-  .dx-popup-content > .dx-template-wrapper.custom-color-picker {
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/DropDownButton/Overview/Vue/App.vue]: why did you remove this style?

File: apps/demos/Demos/DropDownButton/Overview/ReactJs/styles.css [hunk 5]
```diff
@@ -16,3 +16,55 @@
...
+}
+
+.dx-button-row {
+  font-size: 12px;
+  line-height: 14px;
+  opacity: 0.6;
+}
+
+#two-sections .dx-button {
+  min-height: 46px;
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/DropDownButton/Overview/ReactJs/styles.css]: please check if we need min-hight here. i dont think we do

Comment (jdvictoria) [reply apps/demos/Demos/DropDownButton/Overview/ReactJs/styles.css]: You're right it's not necessary now that height has been fixed for every theme. Should I remove other min-height on other frameworks?

File: apps/demos/Demos/DropDownButton/Overview/React/styles.css [hunk 6]
```diff
@@ -16,3 +16,55 @@
...
+  line-height: 20px;
+}
+
+.dx-button-row {
+  font-size: 12px;
+  line-height: 14px;
+  opacity: 0.6;
+}
+
+#two-sections .dx-button {
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/DropDownButton/Overview/React/styles.css]: i suppose we can leace selector as #two-sections ( without .dx-button). But please check it

File: apps/demos/Demos/DropDownButton/Overview/ReactJs/styles.css (hunk 5 above)
Comment (AlexanderMoiseev) [reply apps/demos/Demos/DropDownButton/Overview/ReactJs/styles.css]: yep

File: apps/demos/Demos/DropDownButton/Overview/React/styles.css (hunk 6 above)
Comment (jdvictoria) [reply apps/demos/Demos/DropDownButton/Overview/React/styles.css]: .dx-button is needed to resize the button when other themes are used

File: apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.css [hunk 7]
```diff
@@ -16,3 +16,54 @@
   cursor: pointer;
   font-size: 18px;
 }
+
+.dx-button-wrapper {
```
Comment (EugeniyKiyashko) [inline apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.css]: Before each selector, we should add :ng-deep. See already existing selectors.

File: apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.html [hunk 8]
```diff
@@ -14,25 +14,38 @@
...
-    <div class="dx-field-label"> Custom main button action </div>
+    <div class="dx-field-label">Custom template and actions</div>
     <div class="dx-field-value">
       <dx-drop-down-button
+        id="two-sections"
         [splitButton]="true"
         [useSelectMode]="false"
-        text="Sandra Johnson"
-        icon="../../../../images/gym/coach-woman.png"
+        text="Olyvia Peyton"
```
Comment (EugeniyKiyashko) [inline apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.html]: Seems, it's a typo. In our data we use Name: 'Olivia Peyton' 
image

Comment (jdvictoria) [reply apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.html]: Regarding this, should I follow Olivia (in our data) or in the design (Olyvia)?

Figma - https://www.figma.com/file/792Fik8qn6E3laq9CfFJfW/Components?type=design&node-id=5409-51951&mode=design&t=uSCouvloGwLw7K12-0

File: apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.css [hunk 9]
```diff
@@ -16,3 +16,54 @@
...
+  line-height: 20px;
+}
+
+.dx-button-row {
+  font-size: 12px;
+  line-height: 14px;
+  opacity: 0.6;
+}
+
+::ng-deep #two-sections .dx-button {
```
Comment (EugeniyKiyashko) [inline apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.css]: #two-sections is id of dropDownButton component. I think the id is not accurately describe the component's purpose, functionality, or content, you might consider choosing a more descriptive ID. 
@vladaskorohodova Could you please check class, Ids names that are used?

File: apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.css [hunk 10]
```diff
@@ -16,3 +16,54 @@
...
+  border-style: solid;
+  border-color: var(--dx-color-border);
+  border-radius: 50%;
+}
+
+.button-img-indicator {
+  position: absolute;
+  background-color: var(--dx-color-danger);
+  top: -1px;
+  right: -1px;
```
Comment (EugeniyKiyashko) [inline apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.css]:
```suggestion
  inset-inline-end: -1px;
```

File: apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.html (hunk 8 above)
Comment (EugeniyKiyashko) [reply apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.html]: @vladaskorohodova

Comment (vladaskorohodova) [reply apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.html]: The correct name is Olivia with an "i".

Comment (jdvictoria) [reply apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.html]: Okay, thanks! We'll just fix the tests then refactor it.

File: apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.css (hunk 9 above)
Comment (vladaskorohodova) [reply apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.css]: I would use "container" instead of "wrapper".
button-img-wrapper -> button-img-container
button-wrapper -> text-container

Also, why do some classes have a "dx-" prefix, and some do not? Not consistent.

When it comes to "two-sections", I would rename it to "custom-template".

File: apps/demos/Demos/DropDownButton/Overview/Vue/App.vue [hunk 11]
```diff
@@ -18,26 +18,40 @@
...
+              @button-click="onButtonClick"
+              @item-click="onItemClick"
+              template="dropDownButtonTemplate"
+          >
+            <template #dropDownButtonTemplate="{ data }">
+              <div class="button-img-container">
+                <div class="button-img-indicator"/>
+                <img
+                    class="button-img"
+                    src="/apps/demos/images/employees/51.png"
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/DropDownButton/Overview/Vue/App.vue]: please double check this path

File: apps/demos/Demos/DropDownButton/Overview/jQuery/index.js [hunk 12]
```diff
@@ -20,11 +20,37 @@ $(() => {
     onItemClick(e) {
       DevExpress.ui.notify(e.itemData.name, 'success', 600);
     },
-    text: 'Sandra Johnson',
-    icon: '../../../../images/gym/coach-woman.png',
+    text: 'Olivia Peyton',
+    icon: '../../../../images/employees/51.png',
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/DropDownButton/Overview/jQuery/index.js]: we should not use icon proeprty anymore, we have a template

File: apps/demos/Demos/DropDownButton/Overview/jQuery/index.js [hunk 13]
```diff
@@ -20,11 +20,37 @@ $(() => {
...
+      const $imageWrapper = $("<div>")
+        .addClass('button-img-container').appendTo(element);
+
+      $("<div>")
+        .addClass('button-img-indicator').appendTo($imageWrapper);
+
+      $("<img>")
+        .addClass('button-img')
+        .attr({
+          src: data.icon,
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/DropDownButton/Overview/jQuery/index.js]: it should be a link to an image

File: apps/demos/Demos/DropDownButton/Overview/Vue/App.vue [hunk 14]
```diff
@@ -171,22 +185,78 @@ function onColorClick(clickedColor) {
...
+.demo-container .dx-fieldset:first-child {
+  width: 500px;
+}
+
+.custom-color-picker {
+  width: 82px;
+  padding: 5px;
+}
+
+.dx-button-content img.dx-icon {
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/DropDownButton/Overview/Vue/App.vue]: please check if this style is required, im not sure it affects demo in any way

File: apps/demos/Demos/DropDownButton/Overview/Vue/App.vue (hunk 11 above)
Comment (jdvictoria) [reply apps/demos/Demos/DropDownButton/Overview/Vue/App.vue]: In Vue, the whole component doesn't load when "../../../../images/employees/51.png" is used for the src.

File: apps/demos/Demos/DropDownButton/Overview/jQuery/styles.css [hunk 15]
```diff
@@ -7,16 +7,58 @@
...
-  width: 24px;
-  height: 24px;
-}
-
 .color {
   cursor: pointer;
   font-size: 18px;
 }
 
-#text {
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/DropDownButton/Overview/jQuery/styles.css:20]: why did you remove this?

**Review by AlexanderMoiseev — DISMISSED** [review]

File: apps/demos/Demos/DropDownButton/Overview/jQuery/index.js [hunk 16]
```diff
@@ -20,11 +20,36 @@ $(() => {
...
       DevExpress.ui.notify(e.itemData.name, 'success', 600);
     },
-    text: 'Sandra Johnson',
-    icon: '../../../../images/gym/coach-woman.png',
+    text: 'Olivia Peyton',
     displayExpr: 'name',
     keyExpr: 'id',
     useSelectMode: false,
+    template(data, element) {
+      const $imageWrapper = $("<div>")
```
Comment (vladaskorohodova) [inline apps/demos/Demos/DropDownButton/Overview/jQuery/index.js]:
```suggestion
      const $imageWrapper = $("<div>")
```

Please rename all instances of `imageWrapper` to `imageContainer`.

File: apps/demos/Demos/DropDownButton/Overview/jQuery/index.js [hunk 17]
```diff
@@ -20,11 +20,36 @@ $(() => {
...
+        .addClass('button-img-indicator').appendTo($imageWrapper);
+
+      $("<img>")
+        .addClass('button-img')
+        .attr({
+          src: '../../../../images/employees/51.png',
+          alt: 'employee',
+        }).appendTo($imageWrapper);
+
+      const $wrapper = $("<div>")
```
Comment (vladaskorohodova) [inline apps/demos/Demos/DropDownButton/Overview/jQuery/index.js]:
```suggestion
      const $wrapper = $("<div>")
```

Same with `wrapper`. I would rename it to `textContainer`.

**Review by vladaskorohodova — DISMISSED** [review]

File: apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.html [hunk 18]
```diff
@@ -14,25 +14,41 @@
...
-    <div class="dx-field-label"> Custom main button action </div>
+    <div class="dx-field-label">Custom template and actions</div>
     <div class="dx-field-value">
       <dx-drop-down-button
+        id="custom-template"
         [splitButton]="true"
         [useSelectMode]="false"
-        text="Sandra Johnson"
-        icon="../../../../images/gym/coach-woman.png"
+        text="Olivia Peyton"
```
Comment (AlexanderMoiseev) [inline apps/demos/Demos/DropDownButton/Overview/Angular/app/app.component.html]: we should get rid of text property, like we got rid of icon

**Review by AlexanderMoiseev — APPROVED** [review]

File: apps/demos/Demos/DropDownButton/Overview/React/DropDownButtonTemplate.tsx [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+);
+
+const ButtonDescription = ({ text, description }) => (
+  <div className="text-container">
+    <div className="button-title">{text}</div>
+    <div className="button-row">{description}</div>
+  </div>
+);
+
+const DropDownButtonTemplate = () => (
```
Comment (EugeniyKiyashko) [inline apps/demos/Demos/DropDownButton/Overview/React/DropDownButtonTemplate.tsx:17]:
```suggestion
export default DropDownButtonTemplate = () => (
```

Comment (jdvictoria) [reply apps/demos/Demos/DropDownButton/Overview/React/DropDownButtonTemplate.tsx:17]: I think template doesn't render properly when we try to change const to export default.

**Review by EugeniyKiyashko — APPROVED** [review]
